utils/upload_ansible_roles_to_galaxy.py

```diff
@@ -135,12 +136,55 @@ def gather_data(self):
                     "pre_tasks are not supported for ansible roles and "
                     "will be skipped!.\n")
 
-        self.tasks_local_content = yaml.dump(
-            self.tasks_data, width=120, indent=4, default_flow_style=False)
-
         description = self.get_description_from_filedata(filedata)
         self.description = description
 
+    def add_variables_to_tasks(self):
+        for task in self.tasks_data:
+            if "when" not in task:
+                task["when"] = []
+            elif isinstance(task["when"], str):
```

Comment:
What happens if the task `when` is a list? Eg.
```
when:
  - aaa
  - bbb
```

Reply:
Then, the `task["when"]` is a list, and `add_variables_to_task` will simply append another items to the list.